elf_reader.go

```diff
@@ -575,6 +582,26 @@ func (ec *elfCode) relocateInstruction(ins *asm.Instruction, rel elf.Symbol) err
 			return fmt.Errorf("asm relocation: %s: unsupported type %s", name, typ)
 		}
 
+		// If we arrive here, it can possibly be because we are dealing with
+		// .kconfig.
+		if mapSpec, ok := (*ec.maps)[".kconfig"]; ok {
+			if ds, ok := mapSpec.Value.(*btf.Datasec); ok {
+				for _, varSecInfo := range ds.Vars {
```

Comment:
For consistency with the population counterpart:
```suggestion
				for _, vsi := range m.Value.(*btf.Datasec).Vars {
```
Also, m.Value is guaranteed to be a ds by preceding code.